skcosmo/sparsified/sparse_kernel_pca.py

```diff
@@ -0,0 +1,230 @@
+        values for kernel passed as callable object.
+        Ignored by other kernels.
+
+    fit_inverse_transform : bool, default=False
+        Learn the inverse transform for non-precomputed kernels.
+        (i.e. learn to find the pre-image of a point)
+
+    eigen_solver : {'auto', 'dense', 'arpack'}, default='auto'
+        Select eigensolver to use. If n_components is much less than
+        the number of training samples, arpack may be more efficient
+        than the dense eigensolver.
+
+    tol : float, default=0
+        Convergence tolerance for arpack.
+        If 0, optimal value will be chosen by arpack.
+
+    copy_X : bool, default=True
+        If True, input X is copied and stored by the model in the `X_fit_`
+        attribute. If no further changes will be done to X, setting
+        `copy_X=False` saves memory by storing a reference.
+
+    n_jobs : int, default=None
+        The number of parallel jobs to run.
+        ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors.
+    """
+
+    def __init__(
+        self,
+        n_components,
+        n_active=None,
+        center=True,
+        selector=SampleFPS,
+        selector_args={},
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        fit_inverse_transform=False,
+        tol=0,
+        n_jobs=1,
+        copy_X=True,
+    ):
+        super().__init__(
+            kernel=kernel,
+            gamma=gamma,
+            degree=degree,
+            coef0=coef0,
+            kernel_params=kernel_params,
+            n_active=n_active,
+            n_jobs=n_jobs,
+            selector=selector,
+            selector_args=selector_args,
+        )
+        self.n_components = n_components
+        self.center = center
+        self.tol = tol
+        self._fit_inverse_transform = fit_inverse_transform
+        self.copy_X = copy_X
+
+    def fit(self, X, X_sparse=None, y=None):
+        """Fit the model from data in X.
+
+        :param X: Training vector, where n_samples in the number of samples
+                  and n_features is the number of features. This may also be the
+                  precomputed kernel of shape (n_samples, n_samples)
+                  in the case that self.kernel == 'precomputed'
+        :type X: {array-like} of shape (n_samples, n_features)
+        :param X_sparse: Active set of samples, where n_features is the number of features.
+                         This may also be the precomputed active kernel of shape
+                         (n_active, n_active) in the case that self.kernel == 'precomputed'
+        :type X_sparse: {array-like} of shape (n_active, n_features)
+        :return: Returns the instance itself.
+        """
+        X = check_array(X, copy=self.copy_X)
+
+        K_MM = self._set_sparse_kernel(X, X_sparse)
+        K_NM = self._get_kernel(X, self.X_sparse_)
+
+        if self.center:
```

Comment:
I think, centering should be in the base class. Because it will be common operation for all sparse methods, and it would be more comfortable to write centering 1 time.

Reply:
I see your point -- I'll revert to the former style.

Reply:
Okay, I remember why I did this. I separated generating KMM and KNM into two functions, so that you didn't have to regenerate KMM every time you wanted KNM. This meant that the centerer should live in the space where the two kernels would end up, i.e. the sparse KPCA class.